File: kattis/IX/LTAI-2022-05-02/B/basicprogramming2.cpp
#include <bits/stdc++.h>

using namespace std;

typedef int64_t i64;

int main() {
  i64 n, t;
  cin >> n >> t;
  vector<i64> v(n);
  for (i64 i = 0; i < n; i++) {
    cin >> v[i];
  }

  if (t == 1) {
    map<i64, i64> m;
    for (auto x : v) {
      if (m.count(7777 - x) != 0) {
        cout << "Yes" << endl;
        return 0;
      }
      m[x]++;
    }
    cout << "No" << endl;
  } else if (t == 2) {
    map<i64, i64> m;
    for (i64 i = 0; i < n; i++) {
      m[v[i]]++;
    }
    for (auto x : m) {
      if (x.second > 1) {
        cout << "Contains duplicate" << endl;
        return 0;
      }
    }
    cout << "Unique" << endl;
  } else if (t == 3) {
    map<i64, i64> m;
    for (i64 i = 0; i < n; i++) {
      m[v[i]]++;
    }
    for (auto x : m) {
      if (x.second > n / 2) {
        cout << x.first << endl;
        return 0;
      }
    }
    cout << -1 << endl;
  } else if (t == 4) {
    sort(v.begin(), v.end());
    if (n % 2 == 1) {
      cout << v[n / 2] << endl;
    } else {
      cout << v[n / 2 - 1] << " " << v[n / 2] << endl;
    }
  } else if (t == 5) {
    sort(v.begin(), v.end());
    for (auto x : v) {
      if (x >= 100 && x <= 999) {
        cout << x << " ";
      }
    }
    cout << endl;
  }
}

File: kattis/IX/LTAI-2022-05-09/G/basicprogramming2.py
def main():
    n, t = input().split()
    n = int(n)
    t = int(t)

    v = input().split()
    v = [int(x) for x in v]

    m = {}

    if t == 1:
        for x in v:
            if 7777 - x in m:
                print("Yes")
                return
            elif x not in m:
                m[x] = True

        print("No")

    elif t == 2:
        unique = True
        for x in v:
            if x in m:
                unique = False
                break

            m[x] = True

        if unique:
            print("Unique")
        else:
            print("Contains duplicate")

    elif t == 3:
        for x in v:
            if x not in m:
                m[x] = 0
            m[x] += 1

        for k, v in m.items():
            if v > n // 2:
                print(k)
                return
        print(-1)

    elif t == 4:
        v.sort()
        if n % 2 == 0:
            print(v[n // 2 - 1], end=" ")
        print(v[n // 2])

    elif t == 5:
        v.sort()

        for x in v:
            if x >= 100 and x <= 999:
                print(x, end=" ")


if __name__ == "__main__":
    main()
